pkg/controller/migcluster/validation.go

```diff
@@ -216,6 +226,50 @@ func (r ReconcileMigCluster) testConnection(cluster *migapi.MigCluster) error {
 	return nil
 }
 
+// Validate the Exposed registry route
+func (r ReconcileMigCluster) validateRegistryRoute(cluster *migapi.MigCluster) error {
+	if cluster.Spec.IsHostCluster {
+		return nil
+	}
+
+	if cluster.Status.HasCriticalCondition() {
+		return nil
+	}
+
+	if cluster.Spec.ExposedRegistryPath != "" {
```

Comment:
Just curious:

1. Are we expecting that `cluster.Spec.ExposedRegistryPath` will always be a bare domain name without protocol attached? 
2. Will this work if a non-standard port is in use for registry?

Reply:
The expectation is that it is a host/domain name with an optional port, although I think at one point we added some code that strips off "http[s]://" if it's there since the an early iteration of the UI was adding it in. The path is expected to be used to create a docker image reference `docker://<exposedRegistryPath>/<namespace>/<imagename>:<tag>`
For the internal registry, this will normally have ":5000:" appended. For an exposed route using a normal OpenShift passthrough route, there will be no port specified (I think the actual port in use is 443). If the route is actually created with a non-standard (external) port, then that would be included in the registry path.

Reply:
Should we add validation the protocol is unspecified? Below we also automatically assume the protocol is `https`... isn't it possible for the registry to not be behind `https`?

Reply:
@dymurray So the protocol specified in the generated dockerImageReference is "docker://" -- I'm not quite sure what the internal docker rules are on when to use ssl vs. not there. But logically there is no protocol attached to the exposed path since the path is inserted directly after "docker://" in the generate string.

Reply:
The input value is a "host[:port]" value, not a URL with embedded protocol.

Reply:
But since our test uses https/http rather than "docker://", we need to infer a protocol from the path. If it's possible that there will be an "http://" exposed registry path, we may need to try http when https fails. I don't know that I've run into a scenario where the internal registry uses http, so this may not be a scenario we care about, so assuming that "https://" works may be sufficient here.